### PR TITLE
Fish update just dropped. New fish, aquarium type, two traits and a research node.

### PR DESCRIPTION
**Pull request by tralezab**
#### anxious zipzap
An incredibly anxiety-ridden and electroreceptive fish. Worried about the walls of its tank closing in constantly. Both literally and as a general metaphorical unease about life's direction.

It produces electricity, and HATES being with any other fish. One other zipzap is allowed at most.

image

#### monocloning jumpercable
A surprisingly useful if nasty looking creation from the syndicate fish labs. Drop one in a tank, and watch it self-feed and multiply. Generates more and more power as a growing swarm!

A far better option for power generation than the zipzap, but it's a syndicate fish. Get an emag or get very lucky!

image

Working spacebase example

image

### Aquarium type

Bioelectricity Generator! It boosts the power of electroreceptive fish to very dangerous levels! But, uh, you put a tesla coil, and voila! Free power! ...Free power if you keep the fish fed! More fish, more power!

image

You can convert an aquarium into a bioelectricity generator by researching the new node, "Marine Utility", from bio processing. Then you build the biogen upgrade kit and apply it to an already existing aquarium.

#### Electrogenesis

Trait turns the fish into a shock weapon while it's alive, and every time it feeds, it releases a weak tesla bolt to anyone close to the tank.

#### Anxiety

anxious fish die when any other fish are in the tank. Please keep them alone :[

## Why It's Good For The Game

Fish are great, but fish don't have anything to give back to the station, right? Well, there's some small stuff, but I wanted to add a bit of functionality towards being able to benefit from having fish.

The anxiety trait with the zipzap gives spessmen a bit of a challenge. You can settle for one fish generating power per tank and do a multi-tank generation setup, or you can do some crossbreeding to make a fish WITH electroreceptive but WITHOUT anxiety. Neat!

## Changelog
:cl:
add: New fish, the anxious zipzap
add: New syndicate fish, the monocloning jumpercable
add: New aquarium, the bioelectricity generator
fix: Mixotrophic fish now properly lack food requirements
/:cl:

### CODE REVIEW

**Comment by Higgin** [comment]
Why is it that when virology is on the ropes, fish begin to rise? It's like the fourth time in the last three years.

this is awesome btw, and great quality on the sprites

**Comment by GuillaumePrata** [comment]
Now we remove the SM from Icebox as it has unfixable design issues on that enviroment and replace it with a big frostelectric fish.

Time to save engineering!

File: code/modules/fishing/aquarium/aquarium.dm [hunk 1]
```diff
@@ -212,6 +216,20 @@
...
+			return
+		balloon_alert(user, "upgrading...")
+		if(!do_after(user, 5 SECONDS, src))
+			return
+		var/obj/structure/aquarium/upgraded_aquarium = new upgrade.upgrade_to_type(loc)
+		for(var/atom/movable/moving in contents)
+			moving.forceMove(upgraded_aquarium)
+		balloon_alert(user, "upgraded")
+		qdel(upgrade)
+		qdel(src)
```
Comment (Jacquerel) [inline code/modules/fishing/aquarium/aquarium.dm:232]: presumably we want to return here

File: code/modules/fishing/aquarium/aquarium_upgrades.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	desc = "An unconventional type of generator that boosts and harvests the energy produced by bioelectric fish."
+
+	icon_state = "bioelec_map"
+	icon_prefix = "bioelec"
+
+/obj/structure/aquarium/bioelec_gen/zap_act(power, zap_flags)
+	var/explosive = zap_flags & ZAP_MACHINE_EXPLOSIVE
+	if(!explosive)
+		return //immune to all other shocks to make sure power can be generated without breaking the generator itself
+	. = ..()
```
Comment (Jacquerel) [inline code/modules/fishing/aquarium/aquarium_upgrades.dm:31]:
```suggestion
	return ..()
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fish/fish_traits.dm [hunk 3]
```diff
@@ -379,3 +379,60 @@ GLOBAL_LIST_INIT(fish_traits, init_subtypes_w_path_keys(/datum/fish_trait, list(
...
+		fish.force = fish::force
+		fish.damtype = fish::damtype
+		fish.attack_verb_continuous = fish::attack_verb_continuous
+		fish.attack_verb_simple = fish::attack_verb_simple
+
+///signal sent when the electroreceptive fish is fed, creating a tesla bolt if the fish is in a power generating tank
+/datum/fish_trait/electrogenesis/proc/on_fish_fed(obj/item/fish/fish, seconds_per_tick)
+	SIGNAL_HANDLER
+	var/fish_zap_range = 1
+	var/fish_zap_power = 5000
```
Comment (necromanceranne) [inline code/modules/fishing/fish/fish_traits.dm:431]: Are there not defines for zap strength?

Comment (tralezab) [reply code/modules/fishing/fish/fish_traits.dm:431]: I'd like to use them if there is, but everywhere i looked there wasn't. I just ran the calculations myself. The non-bioelec tank shock is about 8 damage, the bioelec tank shock is 33. Would be nice if we had a define that converted a damage amount to the number required to cause it

Comment (tralezab) [reply code/modules/fishing/fish/fish_traits.dm]: There was joules which kinda helps

**Comment by CCC23** [comment]
time to make fish chernobyle as a tot with 30000 jumper cables  30 fish tanks and enough electricity production to vaporize anyone even near a power cable

File: code/modules/fishing/fish/fish_types.dm [hunk 4]
```diff
@@ -612,3 +632,26 @@
...
+	average_weight = 500
+	random_case_rarity = FISH_RARITY_VERY_RARE
+	favorite_bait = list(/obj/item/food/bait/doughball)
+	required_temperature_min = MIN_AQUARIUM_TEMP+18
+	required_temperature_max = MIN_AQUARIUM_TEMP+26
+	fish_traits = list(
+		/datum/fish_trait/no_mating,
+		/datum/fish_trait/wary,
+		/datum/fish_trait/anxiety,
+		/datum/fish_trait/electrogenesis
```
Comment (san7890) [inline code/modules/fishing/fish/fish_types.dm:654]:
```suggestion
		/datum/fish_trait/electrogenesis,
```

File: code/modules/research/designs/power_designs.dm [hunk 5]
```diff
@@ -101,6 +101,16 @@
 	)
 	departmental_flags = DEPARTMENT_BITFLAG_ENGINEERING
 
+/datum/design/board/bioelec_gen
+	name = "Aquarium Bioelectricity Kit"
+	desc = "The required components to convert an aquarium into a bioelectricity generator."
+	id = "bioelec_gen"
+	build_path = /obj/item/aquarium_upgrade/bioelec_gen
+	category = list(
+		RND_CATEGORY_MACHINE + RND_SUBCATEGORY_MACHINE_ENGINEERING
```
Comment (san7890) [inline code/modules/research/designs/power_designs.dm:110]:
```suggestion
		RND_CATEGORY_MACHINE + RND_SUBCATEGORY_MACHINE_ENGINEERING,
```

File: code/modules/fishing/fish/fish_traits.dm [hunk 6]
```diff
@@ -379,3 +379,53 @@ GLOBAL_LIST_INIT(fish_traits, init_subtypes_w_path_keys(/datum/fish_trait, list(
...
+	RegisterSignal(fish, COMSIG_FISH_LIFE, PROC_REF(on_fish_life))
+
+///signal sent when the anxiety fish is fed, killing it if sharing contents with too many fish.
+/datum/fish_trait/anxiety/proc/on_fish_life(obj/item/fish/fish, seconds_per_tick)
+	SIGNAL_HANDLER
+	var/fish_tolerance = 3
+	if(!fish.loc || fish.status == FISH_DEAD)
+		return
+	for(var/obj/item/other_fish in fish.loc.contents)
+		if(!fish_tolerance)
```
Comment (san7890) [inline code/modules/fishing/fish/fish_traits.dm:401]:
```suggestion
		if(fish_tolerance == 0)
```

kindly do not treat integers as booleans

File: code/modules/fishing/fish/fish_types.dm [hunk 7]
```diff
@@ -295,6 +295,26 @@
...
+	stable_population = 12
+	average_size = 110
+	average_weight = 10000
+	random_case_rarity = FISH_RARITY_GOOD_LUCK_FINDING_THIS
+	required_temperature_min = MIN_AQUARIUM_TEMP+10
+	required_temperature_max = MIN_AQUARIUM_TEMP+30
+	fish_traits = list(
+		/datum/fish_trait/parthenogenesis,
+		/datum/fish_trait/mixotroph,
+		/datum/fish_trait/electrogenesis
```
Comment (san7890) [inline code/modules/fishing/fish/fish_types.dm:315]:
```suggestion
		/datum/fish_trait/electrogenesis,
```

**Review by san7890 — COMMENTED** [review]
bless my beautiful trailing commas in lists life (and other nit)

**Comment by Admiral-Gigaton** [comment]
> time to make fish chernobyle as a tot with 30000 jumper cables 30 fish tanks and enough electricity production to vaporize anyone even near a power cable

not a bad idea... how much fish fits in an aquarium, or one of these electric aquariums?

**Comment by tralezab** [comment]
oh its totally possible with the syndicate fish, if you can get your hands on it. 20 fish maximum in an aquarium, cables multiply up to 12 so you'll need a separate breeding tank to get past 12 soft cap, then you consider multiple tanks and yeaahhhh you can make something hilarious

**Comment by tralezab** [comment]
review handled, ready 👍

**Pull request merged**